### PR TITLE
Assistant: Add VaultPress card to the assistant flow

### PR DESCRIPTION
**Pull request by IanRamosC**
Context: 1201210512993648-as-1203060617058104

## Proposed changes:

This PR changes the Assistant flow for free users. The following changes were added:

* Added two different cards to the Assistant flow: one highlighting VaultPress for free users, the other one more focused on WooCommerce stores
* Updated the e2e tests to cover the changes on the Assistant
* Updated how the list style in `descriptionList` is displayed
* Conditionally render [the description paragraph](https://github.com/Automattic/jetpack/blob/c5d6a4b3b83c8b8ce71a736e852cbd58d172e56d/projects/plugins/jetpack/_inc/client/recommendations/prompts/prompt-layout/index.jsx#L79) to avoid the element and its style if a description is not available
* Added a quick helper function to calculate the monthly price of the product. That will change in the future (see pbNhbs-53E-p2)

Note: the CTA text on the card page (i.e. "Try for $1 for 30 days") is dynamic. It's formatted to your currency settings, and might change if there's an intro offer or not.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pbNhbs-4X5-p2

## Does this pull request change what data or activity we track or use?

No

### Testing the flow for free users

* Spin up an ephemeral/JN site with Jetpack Beta included (so you can test this branch)
* Connect your site
* Choose a free plan on the pricing page
* Continue to the Assistant; make sure the last screen before the Summary is the VaultPress card
* Confirm that the card looks like the screenshots below, and it matches the design (some design changes were not applied to not mess other steps in the Assistant flow)
* Make sure the CTA text shows the right and localized price
  * If you click on the CTA, it will redirect you to the checkout page. After purchase, it will redirect back to the feature onboarding flow
  * If you click "Not now", you should be redirected to recommendations/summary

### Testing the flow for free users who have WooCommerce installed and active
* Follow the test plan described above, but in the first step, spin up a site with both Jetpack Beta **AND** WooCommerce. Make sure to activate WooCommerce before proceeding
* Follow the test above and confirm that the design and copy look like what's in the screenshot below

### Testing the flow for paid users
* The VaultPress Backup card should not show at all. You can test it going through the steps above, but purchasing a product right after connecting Jetpack (on the pricing page)

### Screenshots
|  Without WooCommerce |  With WooCommerce |
|-------------|----------|
| image | image |

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js [hunk 1]
```diff
@@ -547,6 +558,90 @@ export const getStepContent = ( state, stepSlug ) => {
...
+				: sprintf(
+						/* translators: %s: is a formatted currency. e.g. $1 */
+						__( 'Try for %s for 30 days', 'jetpack' ),
+						price
+				  );
+
+			return {
+				progressValue: 100,
+				question: __(
+					'Never lose your site, even if your host goes down (along with your backups)',
```
Comment (CodeyGuyDylan) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:577]: I don't have a specific suggestion to fix this, but the title here seems a bit long. It looks very wordy to me when viewing the recommendation. Could just be me 🤷🏻

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js [hunk 2]
```diff
@@ -115,6 +121,11 @@ export const getSummaryResourceProps = ( state, resourceSlug ) => {
 				ctaLabel: __( 'Add', 'jetpack' ),
 				ctaLink: getJetpackCloudUrl( state, 'settings' ),
 			};
+		case 'vaultpress-backup':
+		case 'vaultpress-for-woocommerce':
+			return {
+				displayName: __( 'VaultPress Backup', 'jetpack' ),
```
Comment (CodeyGuyDylan) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:127]: I believe this is supposed to be what shows up as the display name on the recommendations summary page correct? If we don't plan to show it there we can probably remove this from the switch statement. If we do plan to show it, I believe we'll need to add a `ctaLabel` and `ctaLink` as well

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js [hunk 3]
```diff
@@ -547,6 +558,90 @@ export const getStepContent = ( state, stepSlug ) => {
...
+			const ctaText = isFetchingSiteProducts( state )
+				? __( 'Try for 30 days', 'jetpack' )
+				: sprintf(
+						/* translators: %s: is a formatted currency. e.g. $1 */
+						__( 'Try for %s for 30 days', 'jetpack' ),
+						price
+				  );
+
+			return {
+				progressValue: 100,
```
Comment (CodeyGuyDylan) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:575]: The progress value of 100 here is good since it is the last one, but the previous recommendations `progressValue` will need to be adjusted so it grows at a correct rate

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js [hunk 4]
```diff
@@ -547,6 +558,90 @@ export const getStepContent = ( state, stepSlug ) => {
...
+			const monthlyPrice = getSiteProductMonthlyCost( state, PLAN_JETPACK_BACKUP_T1_YEARLY );
+			const product = getSiteProduct( state, PLAN_JETPACK_BACKUP_T1_YEARLY );
+			const price = formatCurrency( monthlyPrice, product?.currency_code );
+			const ctaText = isFetchingSiteProducts( state )
+				? __( 'Try for 30 days', 'jetpack' )
+				: sprintf(
+						/* translators: %s: is a formatted currency. e.g. $1 */
+						__( 'Try for %s for 30 days', 'jetpack' ),
+						price
+				  );
```
Comment (CodeyGuyDylan) [inline projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:615]: Nice touch adding the placeholder!

File: projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js [hunk 5]
```diff
@@ -547,6 +553,13 @@ export const getProductSlugForStep = ( state, step ) => {
 	return false;
 };
 
+const shouldRecommendVaultPress = ( state, isWooCommerceRequired = false ) => {
+	const sitePlan = getSitePlan( state ).product_slug;
+	const isFree = 'is-free-plan' === getPlanClass( sitePlan );
+
+	return isFree && isWooCommerceActive( state ) === isWooCommerceRequired;
```
Comment (CodeyGuyDylan) [inline projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js:560]: This would imply that if a user has any paid Jetpack plan, not just VaultPress, that this recommendation would not be shown. I think that this is the correct behavior based on what I can find, just wanted to confirm 😄

File: projects/plugins/jetpack/_inc/client/state/site-products/reducer.js [hunk 6]
```diff
@@ -34,18 +34,46 @@ export const reducer = combineReducers( {
...
+/**
+ * Returns the monthly cost of a product. This also takes into account intro offers.
+ *
+ * @see     pbNhbs-53E-p2
+ * @todo    Fix how we calculate the price in the future.
+ * @param   {object} state - Global state tree
+ * @param   {string} slug  - Product slug
+ * @returns {number}  Monthly cost of a product
+ */
+export function getSiteProductMonthlyCost( state, slug ) {
```
Comment (CodeyGuyDylan) [inline projects/plugins/jetpack/_inc/client/state/site-products/reducer.js:74]: I can't remember exactly how they work in the pricing object, but do we need to account for potential sale coupons in this function too? I know this is a temporary solution until we are able to get the pricing where we want it, just a thought I had

File: tools/eslint-excludelist.json [hunk 7]
```diff
@@ -47,7 +47,6 @@
 	"projects/plugins/jetpack/_inc/client/state/search/reducer.js",
 	"projects/plugins/jetpack/_inc/client/state/settings/actions.js",
 	"projects/plugins/jetpack/_inc/client/state/settings/reducer.js",
-	"projects/plugins/jetpack/_inc/client/state/site-products/reducer.js",
```
Comment (CodeyGuyDylan) [inline tools/eslint-excludelist.json:50]: Just wondering why this was excluded before, and what the reason is for not excluding it anymore?

**Review by CodeyGuyDylan — CHANGES_REQUESTED** [review]
When testing the VaultPress card, I noticed the progress bar does not fill up correctly, and instead hits 100% a card before the VaultPress one

https://user-images.githubusercontent.com/65001528/217432140-9c7394fa-914b-43e3-b471-b755d502ffab.mov

Should the VaultPress recommendation be shown in the summary like the others are? I see it missing from the list

image

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js (hunk 1 above)
Comment (IanRamosC) [reply projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:577]: Not just you, but we talked about that here p1HpG7-k7l-p2#comment-60011

File: projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js (hunk 2 above)
Comment (IanRamosC) [reply projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js:127]: Exactly. I think that shouldn't be shown in the summary page - since this doesn't really fit as a resource nor as a feature recommendation. But I can ask the team to hear their opinion on that.

File: projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js (hunk 5 above)
Comment (IanRamosC) [reply projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js:560]: That is correct.

File: tools/eslint-excludelist.json (hunk 7 above)
Comment (IanRamosC) [reply tools/eslint-excludelist.json:50]: Those reducers (and other files in that list) follow an old formatting. The idea of having the exclude list is to not having to fix all those files at once. I removed now because I had to touch it and the linter fixed it. The pre-commit doesn't allow us to push if the file has linting errors, and the CI breaks if the file has no error but it's in that list.

File: projects/plugins/jetpack/_inc/client/state/site-products/reducer.js (hunk 6 above)
Comment (IanRamosC) [reply projects/plugins/jetpack/_inc/client/state/site-products/reducer.js:74]: Good thought. This follows the same formula as the other places where it calculates the price, and IIRC it takes the sale coupons into account.

File: tools/eslint-excludelist.json (hunk 7 above)
Comment (CodeyGuyDylan) [reply tools/eslint-excludelist.json:50]: Good to know! Thanks for explaining

**Review by CodeyGuyDylan — APPROVED** [review]
As long as it's decided here: p1HpG7-k7l-p2#comment-60779 that we don't need to have this on the summary page, all looks good and tests well

**Comment by IanRamosC** [comment]
> As long as it's decided here: p1HpG7-k7l-p2#comment-60779 that we don't need to have this on the summary page, all looks good and tests well

Just to document it here: yes, we don't want to have this on the summary page for now.

**Comment by samiff** [comment]
@IanRamosC This PR tested well for me, but could you verify that the `Never lose your site, even if your host goes down (along with your backups)` heading is what we want here? Seems a bit wordy to me and I see something else mentioned on the P2 post.

**Comment by CodeyGuyDylan** [comment]
> @IanRamosC This PR tested well for me, but could you verify that the `Never lose your site, even if your host goes down (along with your backups)` heading is what we want here? Seems a bit wordy to me and I see something else mentioned on the P2 post.

FWIW I also [asked this question](https://github.com/Automattic/jetpack/pull/28741#discussion_r1099634609) 😄

**Review by samiff — APPROVED** [review]
